In [1]:
import pandas as pd

In [10]:
google = "/content/GOOGL.csv"
goog = pd.read_csv(google)
goog.head()

Date       Open       High        Low      Close  Adj Close    Volume
0  2004-08-19  50.050049  52.082081  48.028027  50.220219  50.220219  44659096
1  2004-08-20  50.555557  54.594597  50.300301  54.209209  54.209209  22834343
2  2004-08-23  55.430431  56.796799  54.579578  54.754753  54.754753  18256126
3  2004-08-24  55.675674  55.855858  51.836838  52.487488  52.487488  15247337
4  2004-08-25  52.532532  54.054054  51.991993  53.053055  53.053055   9188602

In [11]:
goog.shape

(4431, 7)

**First Choice: Google Stock**

1) Source of data

https://www.kaggle.com/datasets/varpit94/google-stock-data/versions/6?resource=download


2) Brief description of data

The dataset provided reflects Google's stock value from 2004 to 2022. There are 7 columns of data: Date, Open, High, Low, Close, Adjusted Close, and Volume. 

3) What is the target?

I will be targeting the High in the dataset to determine the best time to sell the stock.

4) Is this a classification or regression problem?

The dataset is a regression problem.

5) How many features?

There will be 6 features: Date, Open, Low, Close, Adjusted Close, and Volume.

6) How many rows of data.

There are 4,431 rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

Due to stocks being difficult to predict, I believe modeling the dataset will be a big challenge. I will attempt to predict the future high value of Google's stock only using the data provided.

In [12]:
bitcoin = "/content/coin_Bitcoin.csv"
btc = pd.read_csv(bitcoin)
btc.head()

SNo     Name Symbol                 Date        High         Low  \
0    1  Bitcoin    BTC  2013-04-29 23:59:59  147.488007  134.000000   
1    2  Bitcoin    BTC  2013-04-30 23:59:59  146.929993  134.050003   
2    3  Bitcoin    BTC  2013-05-01 23:59:59  139.889999  107.720001   
3    4  Bitcoin    BTC  2013-05-02 23:59:59  125.599998   92.281898   
4    5  Bitcoin    BTC  2013-05-03 23:59:59  108.127998   79.099998   

         Open       Close  Volume     Marketcap  
0  134.444000  144.539993     0.0  1.603769e+09  
1  144.000000  139.000000     0.0  1.542813e+09  
2  139.000000  116.989998     0.0  1.298955e+09  
3  116.379997  105.209999     0.0  1.168517e+09  
4  106.250000   97.750000     0.0  1.085995e+09

In [13]:
btc.shape

(2991, 10)

**Second Choice: Bitcoin Stock**

1) Source of data

https://www.kaggle.com/datasets/sudalairajkumar/cryptocurrencypricehistory?select=coin_Bitcoin.csv

2) Brief description of data

The dataset provided reflects Bitcoin's stock value from 2013 to 2021. There are 10 columns of data: Name, Symbol, Date, Open, High, Low, Close, Adjusted Close, Volume, and Market Cap.

3) What is the target?

I will be targeting the High in the dataset to determine the best time to sell the stock.

4) Is this a classification or regression problem?

The dataset is a regression problem.

5) How many features?

There will be 7 features: Date, Open, Low, Close, Adjusted Close, Volume, and Market Cap.

6) How many rows of data.

There are 2,991 rows of data.

7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?

Due to cryptocurrency being thought to be unpredictable, I believe modeling the dataset will be a big challenge. I will attempt to predict the future high value of Bitcoin's stock only using the data provided.